firmware/Core/Src/mpi/mpi_command_handling.c

```diff
@@ -58,61 +39,62 @@ uint8_t MPI_send_telecommand_hex(const uint8_t *bytes_to_send, size_t bytes_to_s
 	
 	// Check for UART reception errors
     if (receive_status != HAL_OK) {
-        return 2; // Error code: Failed UART reception
+        return 3; // Error code: Failed UART reception
     }
 
+	// Record start time for mpi response reception
 	uint32_t UART_mpi_rx_start_time_ms = HAL_GetTick();
 
 	while(1){
 		// MPI hasn't sent any data and has timed out
-		if(((HAL_GetTick() - UART_mpi_rx_start_time_ms) > timeout_duration_ms) && (UART_mpi_rx_buffer_write_idx == 0)){
-			DEBUG_uart_print_str("Break1\n");
-			break; // TODO: handle case
+		if(((HAL_GetTick() - UART_mpi_rx_start_time_ms) > RX_TIMEOUT_DURATION_MS) && (UART_mpi_rx_buffer_write_idx == 0)){
```

Comment:
Why isn't all the end-of-receive stuff (like cancelling the DMA) done here also? Suggest refactoring.